apps/chef_objects/src/chef_user.erl

```diff
@@ -347,8 +363,12 @@ update_from_ejson(#chef_user{} = User, UserEJson) ->
     end,
     Name = username_from_ejson(UserEJson),
     Email = ej:get({<<"email">>}, UserEJson),
-    ExternalAuthenticationUid = value_or_null({<<"external_authentication_uid">>}, UserEJson),
-    RecoveryAuthenticationEnabled = ej:get({<<"recovery_authentication_enabled">>}, UserEJson) =:= true,
```

Comment:
@marcparadise Well, let's definitely make this behave like we want while we're in here.  However, I'm not sure resetting it on login is what people would want either.  This is speculation since I've don't know how most people use this feature, but I would probably use it like this:

1) Set a recovery password for my most trusted users.
2) Use chef normally for months
3) When the LDAP server goes down, try to fix it.  If we need to use Chef via the webui in the meantime, have one of the trusted users from (1) do it.

Resetting it on login would mean that you would have to explicitly reset the recovery password every time you needed to use it (assuming that the user logged in normally between those times).